### PR TITLE
PBI-31-32: Edit & Delete Comment in Lesson Material

### LINKED ISSUE #72: PBI-31: Creator can edit comment on the lesson material page
**Story Points: 8**

- [x] The edit button can only be seen by the creator
- [x] Current user id and comment created by id user must be the same
- [x] Content field (Max 65535) (Required)

### LINKED ISSUE #73: PBI-32: Teacher & Creator can delete comment on the lesson material page
**Story Points: 8**

- [x] The delete button can only be seen by the creator and the teacher
- [x] Current user id and comment created by id user must be the same OR 
- [x] Current user id and teacher id must be the same
- [x] Pass lesson material ID to Back End

### PR DESCRIPTION
**Pull request by IlmaMannix**
**Student POV**
image

**Teacher POV**
image

**Teacher POV (student comment)**
image

### CODE REVIEW

File: src/resources/api/course/materialComment.js [hunk 1]
```diff
@@ -19,3 +19,43 @@ export const createMaterialComment = async (payload) => {
...
+        const response = await api.delete(apiUrl);
+
+        if (!!response.errors) {
+            throw response.errors;
+        }
+
+        data = response.data;
+    } catch (e) {
+        error = e;
+        errorMessage = "Something went wrong, please try again later";
```
Comment (nofamex) [inline src/resources/api/course/materialComment.js:57]: i think you can use error message given by the api

File: src/resources/api/course/materialComment.js [hunk 2]
```diff
@@ -19,3 +19,43 @@ export const createMaterialComment = async (payload) => {
...
+        const response = await api.put(apiUrl, payload);
+
+        if (!!response.errors) {
+            throw response.errors;
+        }
+
+        data = response.data;
+    } catch (e) {
+        error = e;
+        errorMessage = "Something went wrong, please try again later";
```
Comment (nofamex) [inline src/resources/api/course/materialComment.js:37]: use error message given by the api

**Review by nofamex — APPROVED** [review]
LGTM, just a little bit of minor review

**Comment by kanziebub** [comment]
LGTM thank you uwu!

**Review by kanziebub — APPROVED** [review]

**Pull request merged**